From: Benjamin Drung <benjamin.drung@canonical.com>
Date: Tue, 8 Jul 2025 17:58:18 +0200
Subject: fix(lsinitrd): resolve initrd to real path

`lsinitrd /boot/initrd.img` fails in case the initrd is zstd compressed:

```
$ dracut --zstd --no-early-microcode --force
$ 3cpio --examine /boot/initrd.img
0	zstd
$ lsinitrd /boot/initrd.img
Image: /boot/initrd.img: 0
========================================================================
Warning : /boot/initrd.img is a symbolic link, ignoring
Version:

Arguments:
dracut modules:
Warning : /boot/initrd.img is a symbolic link, ignoring
Warning : /boot/initrd.img is a symbolic link, ignoring
========================================================================
========================================================================
cpio: premature end of archive

dracut cmdline:
Warning : /boot/initrd.img is a symbolic link, ignoring
```

The `CAT` variable is `zstd -d -c`. The zstd man page says about the
`--force` parameter: "disable input and output checks. Allows [...]
operating on links". So this zstd behavior is intentional.

Resolve the initrd path to the real path. Then `lsinitrd` prints the
correct content and image size:

```
$ lsinitrd /boot/initrd.img
Image: /boot/initrd.img-6.14.0-23-generic: 15M
[...]
```

Bug-Ubuntu: https://launchpad.net/bugs/2116183
Forwarded: https://github.com/dracut-ng/dracut-ng/pull/1426
---
 lsinitrd.sh | 1 +
 1 file changed, 1 insertion(+)

diff --git a/lsinitrd.sh b/lsinitrd.sh
index ea26a26..688d659 100755
--- a/lsinitrd.sh
+++ b/lsinitrd.sh
@@ -173,6 +173,7 @@ if ! [[ -f $image ]]; then
     usage
     exit 1
 fi
+image=$(realpath "$image")
 
 TMPDIR="$(mktemp -d -t lsinitrd.XXXXXX)"
 # shellcheck disable=SC2064
